crates/relayer-cli/src/config.rs

```diff
@@ -87,15 +87,18 @@ pub fn validate_config(config: &Config) -> Result<(), Diagnostic<Error>> {
     // Check for duplicate chain configuration and invalid trust thresholds
     let mut unique_chain_ids = BTreeSet::new();
     for c in config.chains.iter() {
-        let already_present = !unique_chain_ids.insert(c.id.clone());
+        let already_present = !unique_chain_ids.insert(c.id().clone());
         if already_present {
-            return Err(Diagnostic::Error(Error::duplicate_chains(c.id.clone())));
+            return Err(Diagnostic::Error(Error::duplicate_chains(c.id().clone())));
         }
 
-        validate_trust_threshold(&c.id, c.trust_threshold)?;
-
-        // Validate gas-related settings
-        validate_gas_settings(&c.id, c)?;
+        // TODO: why are these not methods on the config object?
+        // why are they defined only in the relayer-cli crate?
+        if let ChainConfig::CosmosSdk(cosmos_c) = c {
+            validate_trust_threshold(&cosmos_c.id, cosmos_c.trust_threshold)?;
+            // Validate gas-related settings
+            validate_gas_settings(&cosmos_c.id, cosmos_c)?;
+        }
```

Comment:
Shouldn't config validation code be attached to the configs in the core crate, rather than being in the cli? Happy to move it if that sounds good.